packages/starlight/utils/starlight-page.ts

```diff
@@ -211,5 +211,13 @@ async function getUserDocsSchema(): Promise<
 	NonNullable<ContentConfig['collections']['docs']['schema']>
 > {
 	const userCollections = (await import('virtual:starlight/collection-config')).collections;
-	return userCollections?.docs.schema ?? docsSchema();
+  const userSchema = userCollections?.docs?.schema;
+
+  if (typeof userSchema === 'function') {
+    return userSchema;
+  } else if (userSchema) {
+    return () => userSchema;
+  } else {
+    return docsSchema();
+  }
```

Comment:
Wouldn’t this be the simpler fix?
```suggestion
  return userCollections?.docs?.schema ?? docsSchema();
```

Reply:
This needs a UT for help.